content/blog/automate-boot-from-san-target-configurations-using-redfish.md

```diff
@@ -0,0 +1,323 @@
+* QLogic SN1610Q cards do not support the steps outlined in this document (as of April 2023) However, future firmware releases might enable this Redfish API based configuration method.
+
+# Pre-requisite information
+
+Before we dive into the steps for performing the automation, you will need to gather a few pieces of information, such as
+
+* HPE iLO Management port ip address and credentials
+* Storage target World Wide Number (WWN) and LUN id.  A storage administrator can typically provide you with this information.
+
+Download and install the [HPE RESTful Interface Tool](https://www.hpe.com/info/resttool)  (version 4.1.0.0 or later) on the system you will initiate the automation actions from.
+Once the RESTful Interface tool (referred to as iLOrest in the remainder of the document), you can review the current boot target and their order of execution by following the steps below
+
+`ilorest login <ilo-ip> -u <ilo-user> -p password`
+
+`ilorest bootorder`
+
+Refer to the [iLOrest documentation](https://hewlettpackard.github.io/python-redfish-utility/#bootorder-command) for more information on this command.
+
+# Configuration Steps
+
+Now that you have gathered the pre-requisite information and installed the ilorest tool, let’s discuss the detailed steps for implementing the automation.
+
+## Step 1: Ensure that the `FCScanPolicy` value is set to ‘AllTargets’
+
+```markdown
+ilorest  get FCScanPolicy --selector Bios
+```
+
+Expected output:
+
+```markdown
+FCScanPolicy=AllTargets
+```
+
+If the value is not set to AllTargets, you can change it using the following:
+
+```markdown
+ilorest set "FCScanPolicy"="AllTargets" --selector Bios
+
+ilorest commit
+
+ilorest reboot
+```
+
+**Note**:
+
+A﻿fter restarting the server, remember to logout/login again to the server using ilorest tool to retrieve the updated settings from the server directly, rather than the cache.  You can also update the ilorest cache contents with the following command
+
+```markdown
+ilorest select Bios. --refresh
+```
+
+You can monitor the server restart state and continue the configuration when it reaches the `FinishedPost` state, and the `vMainDeviceDiscoveryComplete` device discovery state.  More details about monitoring server state is described in this [article](https://developer.hpe.com/blog/master-the-redfish-server-states-to-improve-your-monitoring-and-manageme/).
+
+The server state can be retrieved with
+
+```markdown
+ilorest serverstate
+```
+
+``
+
+The device discovery state can be obtained with
+
+```markdown
+ilorest get Oem/Hpe/DeviceDiscoveryComplete/DeviceDiscovery --json
+```
+
+After restarting the server, you can verify that the settings have taken effect using the by navigating to the BIOS menus as shown in Figure 1 or using the commands
+
+```
+ilorest get FCScanPolicy --selector Bios
```

Comment:
To be on the safe side, replace line 91 with the following two lines:

ilorest select Bios. --refresh
ilorest get FCScanPolicy